src/uipath/_cli/_utils/_service_cli_generator.py

```diff
@@ -0,0 +1,705 @@
+
+        Args:
+            client: Client instance with service attributes
+
+        Raises:
+            TypeError: If service doesn't implement CRUDServiceProtocol
+            AttributeError: If service attribute doesn't exist on client
+
+        Example:
+            >>> generator = ServiceCLIGenerator(metadata)
+            >>> from uipath import UiPath
+            >>> client = UiPath(...)
+            >>> generator.validate_service(client)  # Fails early if invalid
+            >>> generator.register(cli_group)  # Now safe to register
+        """
+        service = getattr(client, self.metadata.service_attr)
+        validate_service_protocol(service, self.metadata.service_name)
+
+    def register(self, parent_group: click.Group) -> click.Group:
+        """Register service commands under a parent Click group.
+
+        This creates a new Click group for the service and adds all
+        CRUD commands to it based on the metadata.
+
+        Args:
+            parent_group: Parent Click group to add service group to
+
+        Returns:
+            The created service Click group
+
+        Raises:
+            TypeError: If service doesn't implement CRUDServiceProtocol
+
+        Example:
+            >>> @click.group()
+            ... def cli():
+            ...     pass
+            >>> generator = ServiceCLIGenerator(metadata)
+            >>> service_group = generator.register(cli)
+        """
+        service_group = click.Group(
+            name=self.metadata.service_name,
+            help=f"Manage {self.metadata.resource_plural}",
+        )
+
+        self._add_list_command(service_group)
+        self._add_retrieve_command(service_group)
+        self._add_create_command(service_group)
+        self._add_delete_command(service_group)
+
+        self._add_exists_command(service_group)
+
+        parent_group.add_command(service_group)
+
+        return service_group
+
+    def _add_list_command(self, group: click.Group) -> None:
+        """Add list command to the group.
+
+        Generates: uipath <service> list [--folder-path PATH] [--folder-key KEY] [--limit N] [--offset N]
+        """
+
+        @group.command("list")
+        @click.option("--limit", type=int, help="Maximum number of items to return")
+        @click.option("--offset", type=int, default=0, help="Number of items to skip")
+        @common_service_options
+        @service_command
+        def list_cmd(
+            ctx: click.Context,
+            folder_path: str | None,
+            folder_key: str | None,
+            format: str | None,
+            output: str | None,
+            limit: int | None,
+            offset: int,
+        ) -> list[Any]:
+            """List resources."""
+            client = ServiceCommandBase.get_client(ctx)
+            service = getattr(client, self.metadata.service_attr)
+
+            resources = service.list(folder_path=folder_path, folder_key=folder_key)
```

Comment:
I think it would be nice if we handle 401 and instruct the user to use `uipath auth`:
```
(uipath-python) PS C:\Users\cristian.pufu\Documents\GitHub\uipath-python> uipath buckets list
Error: 
Request URL: https://alpha.uipath.com/Ada/byoa/orchestrator_/odata/Buckets?%24skip=0&%24top=100
HTTP Method: GET
Status Code: 401
Response Content: No content
```

Reply:
Same for missing folder key/path env variable:
```
(uipath-python) PS C:\Users\cristian.pufu\Documents\GitHub\uipath-python> uipath buckets list
Error: 
Request URL: https://alpha.uipath.com/Ada/byoa/orchestrator_/odata/Buckets?%24skip=0&%24top=100
HTTP Method: GET
Status Code: 400
Response Content: {"message":"An organization unit is required for this action.","errorCode":1101,"traceId":"00-23bbae4bd8241e91a0a1ea7ef096d409-2772c8eeabe738e9-01"}
```